devito/ir/equations/algorithms.py

```diff
@@ -95,12 +95,20 @@ def lower_exprs(expressions, **kwargs):
                   for f in retrieve_functions(expr)}
 
         # Handle Indexeds (from index notation)
-        for i in retrieve_indexed(expr):
+        for i in retrieve_indexed(expr, deep=True):
             f = i.function
 
             # Introduce shifting to align with the computational domain
             indices = [(a + o) for a, o in zip(i.indices, f._size_nodomain.left)]
 
+            # Indexify indices of nested functions
+            tmp_mapper = {}
+            for index in indices:
+                for nested_func in retrieve_functions(index):
```

Comment:
so you're retrieving functions inside indexeds?

Reply:
perhaps you may use `retrieve_function_carriers` which returns both functions and indexeds

Reply:
also, unless strictly necessary, use short variable names with no underscore (not always possible obv)